xarray/tests/test_coding_times.py

```diff
@@ -1167,3 +1167,11 @@ def test_decode_0size_datetime(use_cftime):
         use_cftime=use_cftime,
     )
     np.testing.assert_equal(expected, actual)
+
+
+@requires_cftime
+def test_scalar_unit() -> None:
+    # test that a scalar units (often NaN when using to_netcdf) does not raise an error
```

Comment:
Does Xarray actually write a `nan` for units? That seems problematic. Can you open a new issue witha  reproducible example?

Reply:
Opened: #7093